src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -458,6 +458,41 @@ def python_type_to_snow_type(tp: Union[str, Type]) -> Tuple[DataType, bool]:
     raise TypeError(f"invalid type {tp}")
 
 
+def data_type_to_python_type(dtype: DataType) -> Type:
```

Comment:
We already have https://github.com/snowflakedb/snowpark-python/blob/4bc70637f4486a67b34036ccd0b8d523db837225/src/snowflake/snowpark/_internal/type_utils.py#L386, maybe align their names?